controller/hybridgateway/route/httproute_test.go

```diff
@@ -0,0 +1,424 @@
+							{Name: "gateway-2", Namespace: lo.ToPtr(gwtypes.Namespace("custom-namespace"))},
+						},
+					},
+					Rules: []gwtypes.HTTPRouteRule{
+						{
+							BackendRefs: []gwtypes.HTTPBackendRef{
+								{
+									BackendRef: gwtypes.BackendRef{
+										BackendObjectReference: gwtypes.BackendObjectReference{
+											Name: "service-1",
+										},
+									},
+								},
+								{
+									BackendRef: gwtypes.BackendRef{
+										BackendObjectReference: gwtypes.BackendObjectReference{
+											Name:      "service-2",
+											Namespace: lo.ToPtr(gwtypes.Namespace("service-namespace")),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			setupSharedStatus: func(statusMap *SharedRouteStatusMap) {
+				key1 := StatusMapKey(HTTPRouteKey, "test-namespace/test-route", "test-namespace/gateway-1")
+				key2 := StatusMapKey(HTTPRouteKey, "test-namespace/test-route", "custom-namespace/gateway-2")
+
+				statusMap.SharedStatus[key1] = SharedRouteStatus{
+					Services: map[string]ServiceControllerStatus{
+						"test-namespace/service-1":    {ServiceControllerInit: true, ProgrammedBackends: 1},
+						"service-namespace/service-2": {ServiceControllerInit: true, ProgrammedBackends: 1},
+					},
+				}
+				statusMap.SharedStatus[key2] = SharedRouteStatus{
+					Services: map[string]ServiceControllerStatus{
+						"test-namespace/service-1":    {ServiceControllerInit: true, ProgrammedBackends: 1},
+						"service-namespace/service-2": {ServiceControllerInit: true, ProgrammedBackends: 1},
+					},
+				}
+			},
+			expectedConditions: map[string]expectedCondition{
+				"test-namespace/gateway-1": {
+					conditionType:   ConditionTypeBackendsProgrammed,
+					conditionStatus: metav1.ConditionTrue,
+					conditionReason: ConditionReasonBackendsProgrammed,
+				},
+				"custom-namespace/gateway-2": {
+					conditionType:   ConditionTypeBackendsProgrammed,
+					conditionStatus: metav1.ConditionTrue,
+					conditionReason: ConditionReasonBackendsProgrammed,
+				},
+			},
+			expectedEnforceResult: op.Updated,
+		},
+		{
+			name: "ParentRefs with empty (nil) namespace",
+			route: func() gwtypes.HTTPRoute {
+				route := gwtypes.HTTPRoute{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "panic-test-route",
+						Namespace: "test-namespace",
+					},
+					Spec: gwtypes.HTTPRouteSpec{
+						CommonRouteSpec: gwtypes.CommonRouteSpec{
+							ParentRefs: []gwtypes.ParentReference{
+								{Name: "test-gateway"}, // This has nil Namespace
+							},
+						},
+						Rules: []gwtypes.HTTPRouteRule{
+							{
+								BackendRefs: []gwtypes.HTTPBackendRef{
+									{
+										BackendRef: gwtypes.BackendRef{
+											BackendObjectReference: gwtypes.BackendObjectReference{
+												Name: "test-service",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				}
+				// Set up existing parent status with non-nil namespace to trigger the panic
```

Comment:
I'm not sure I follow: we're setting this to trigger panic but does the panic actually happen? Where should it be caught?

Reply:
Sorry, it's a leftover that used to trigger panic, but PR includes a fix. I'm removing